### PR TITLE
chore(ci): install binary cppcheck from snap for cppcheck-daily

### PR DESCRIPTION
**Pull request by veqcc**
## Description

Install bainary cppcheck via snap for cppcheck-daily.
https://snapcraft.io/cppcheck

## Tests performed

Wait till the next daily cppcheck CI works.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: .github/workflows/cppcheck-daily.yaml [hunk 1]
```diff
@@ -16,20 +16,12 @@ jobs:
       - name: Install dependencies
         run: |
           sudo apt-get update
-          sudo apt-get install -y build-essential cmake git libpcre3-dev
+          sudo apt-get install -y git
```
Comment (HansRobo) [inline .github/workflows/cppcheck-daily.yaml:19]: Maybe, we can delete whole of this step

Comment (veqcc) [reply .github/workflows/cppcheck-daily.yaml:19]: Agreed :+1:

Comment (veqcc) [reply .github/workflows/cppcheck-daily.yaml:19]: @HansRobo 
Do you think we can skip even `sudo apt-get update` ?
In other words, all `Install dependencies` steps

Comment (HansRobo) [reply .github/workflows/cppcheck-daily.yaml:19]: Yes, we can.
My opening statement means the deletion of whole of  `Install dependencies` steps.

**Review by HansRobo — APPROVED** [review]

**Pull request merged**